pkg/storage/unified/resource/broadcaster.go

```diff
@@ -124,11 +124,12 @@ func (b *broadcaster[T]) Subscribe(ctx context.Context) (<-chan T, error) {
 
 	// create the subscription
 	sub := make(chan T, 100)
-
 	select {
 	case <-ctx.Done(): // client canceled
+		close(sub)
```

Comment:
Closing the channel is just a signal for caller that no more items will arrive on it. We don't need to close channel, if it's never returned to caller (when returning `nil` from this function).